File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[])
{
    // Error checking
    if (argc != 2)
    {
        printf("Liar liar pants on fire!\n");
        return 1;
    }
    else
    {

        printf("plaintext: ");
        string plaintext = get_string();
        char ciphertext[strlen(plaintext)];

        int key = atoi(argv[1]);

        for (int i = 0, n = strlen(plaintext); i < n; i++)
        {
            if ('a' <= plaintext[i] && plaintext[i] <= 'z')
            {
                ciphertext[i] = (((int) plaintext[i] - 96 + key) % 26) + 96;
            }
            else if ('A' <= plaintext[i] && plaintext[i] <= 'Z')
            {
                ciphertext[i] = (char) (((int) plaintext[i] - 64 + key) % 26) + 64;
            }
            // If special character, save as is
            else
            {
                ciphertext[i] = plaintext[i];
            }
        }
        printf("ciphertext: %s\n",ciphertext);
        return 0;
    }
}

File: pset6/sentimental/caesar.py
#
# Returns scrambled message using caesar's algorithm
#

import sys
from cs50 import get_string

def get_(prompt):
    """Prompt user for positive integer"""
    while True:
        n = get_int(prompt)
        if 0 <= n and n < 24:
            break
    return n

def main():


    # error checking
    if len(sys.argv) != 2:
        print("Liar liar pants on fire!")
        sys.exit(1)


    else:
        plaintext = get_string("plaintext: ")
        key = int(sys.argv[1])
        ciphertext = ''

        for c in plaintext:

            # lowercase
            if (ord('a') <= ord(c) and ord(c) <= ord('z')):
                ciphertext = ciphertext + chr(((ord(c) - 96 + key) % 26) + 96)

            # uppercase
            elif (ord('A') <= ord(c) and ord(c) <= ord('Z')):
                ciphertext = ciphertext + chr(((ord(c) - 64 + key) % 26) + 64)

            else:
                ciphertext = ciphertext + c

        print(f"ciphertext: {ciphertext}")
        sys.exit(0)



if __name__ == "__main__":
    main()

